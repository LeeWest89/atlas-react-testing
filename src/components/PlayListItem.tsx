import { Song } from "./MusicPlayer";

interface listItemProp {
  song: Song;
  isSelected: boolean;
  onSelect: () => void;
}

export default function PlayListItem ({ song, isSelected, onSelect }: listItemProp) {
  return (
    <li onClick={onSelect} className={`w-full h-10 mb-[0.25rem] flex items-center justify-between ${isSelected ? 'bg-lightning-blue': 'hover:bg-lightning-blue'} transition-colors duration-300`}>
      <div className="flex flex-col text-left">
        <p className="font-medium text-sm text-deep-yellow">{song.title}</p>
        <p className="font-medium text-sm text-wing-red">{song.artist}</p>
      </div>
      <div>
        <p className="font-medium text-sm text-wing-red">{song.duration}</p>
      </div>
    </li>
  );
}