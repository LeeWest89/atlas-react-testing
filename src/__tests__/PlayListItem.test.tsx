import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem";

//  Test PlayListItem is selected
test("Selected PlayListItem renders correctly", () => {
  const { container } = render (
    <PlayListItem song={{ id: 2, title: "Tidal Drift", artist: "Echoes of the Sea", genre:"Ambient", duration: "8:02", cover: "https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn6.webp" }} isSelected={true} onSelect={() => {}} />
  );
  expect(container).toMatchSnapshot();
});

// Test PlayListItem is unselected
test("Unselected PlayListItem renders correctly", () => {
  const { container } = render (
    <PlayListItem song={{ id: 2, title: "Tidal Drift", artist: "Echoes of the Sea", genre:"Ambient", duration: "8:02", cover: "https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn6.webp" }} isSelected={false} onSelect={() => {}} />
  );
  expect(container).toMatchSnapshot();
});
