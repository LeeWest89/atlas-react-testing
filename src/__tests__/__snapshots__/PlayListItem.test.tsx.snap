// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Selected PlayListItem renders correctly 1`] = `
<div>
  <li
    class="w-full h-10 mb-[0.25rem] flex items-center justify-between bg-lightning-blue transition-colors duration-300"
  >
    <div
      class="flex flex-col text-left"
    >
      <p
        class="font-medium text-sm text-deep-yellow"
      >
        Tidal Drift
      </p>
      <p
        class="font-medium text-sm text-wing-red"
      >
        Echoes of the Sea
      </p>
    </div>
    <div>
      <p
        class="font-medium text-sm text-wing-red"
      >
        8:02
      </p>
    </div>
  </li>
</div>
`;

exports[`Unselected PlayListItem renders correctly 1`] = `
<div>
  <li
    class="w-full h-10 mb-[0.25rem] flex items-center justify-between hover:bg-lightning-blue transition-colors duration-300"
  >
    <div
      class="flex flex-col text-left"
    >
      <p
        class="font-medium text-sm text-deep-yellow"
      >
        Tidal Drift
      </p>
      <p
        class="font-medium text-sm text-wing-red"
      >
        Echoes of the Sea
      </p>
    </div>
    <div>
      <p
        class="font-medium text-sm text-wing-red"
      >
        8:02
      </p>
    </div>
  </li>
</div>
`;
